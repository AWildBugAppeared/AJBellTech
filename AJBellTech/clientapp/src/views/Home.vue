<template>
  <div>
    <h1 class="pa-8 white--text">AJ Bell Investment Platform</h1>
    <Ticker></Ticker>
    <TickerCurrencyConverter></TickerCurrencyConverter>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

import Ticker from "@/components/Ticker.vue";
import TickerCurrencyConverter from "@/components/TickerCurrencyConverter.vue";

export default Vue.extend({
  name: "Home",
  components: { Ticker, TickerCurrencyConverter },
});
</script>
