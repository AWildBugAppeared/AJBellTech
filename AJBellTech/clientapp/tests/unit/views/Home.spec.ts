/* eslint-disable  @typescript-eslint/no-explicit-any */

import Vue from "vue";
import Vuetify from "vuetify";
import { createLocalVue, mount, Wrapper } from "@vue/test-utils";

import Home from "@/views/Home.vue";

describe("Home.vue", () => {
  const localVue = createLocalVue();
  let wrapper: Wrapper<Vue & { [key: string]: any }>;
  let mountWrapper: () => void;

  beforeEach(() => {
    const vuetify = new Vuetify({
      mocks: {
        $vuetify: {},
      },
    });
    Vue.use(Vuetify);
    document.body.setAttribute("data-app", "true");

    mountWrapper = () => {
      wrapper = mount(Home, {
        localVue,
        vuetify,
        data() {
          return {};
        },
        stubs: ['Ticker', 'TickerCurrencyConverter'],
      });
    };
  });

  afterEach(() => {
    wrapper.destroy();
    jest.resetModules();
    jest.clearAllMocks();
  });

  describe("rendering", () => {
    it("should render components", async () => {
      mountWrapper();
      await wrapper.vm.$nextTick();
      await wrapper.vm.$nextTick();

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
