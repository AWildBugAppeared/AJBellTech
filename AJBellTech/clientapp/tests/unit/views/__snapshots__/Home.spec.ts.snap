// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue rendering should render components 1`] = `
<div>
  <h1 class="pa-8 white--text">AJ Bell Investment Platform</h1>
  <ticker-stub></ticker-stub>
  <tickercurrencyconverter-stub></tickercurrencyconverter-stub>
</div>
`;
